### PR TITLE
[WFLY-15349] Use batch statement when creating ejb timer table in DatabaseTimerPersistence

### PR DESCRIPTION
**Pull request by petrberan**
https://issues.redhat.com/browse/WFLY-15349

### CODE REVIEW

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java [hunk 1]
```diff
@@ -398,13 +398,14 @@ private void checkDatabase() {
...
-                        } finally {
-                            safeClose(statement);
+                    try {
+                        statement = connection.createStatement();
+                        for (final String sql : statements) {
+                            statement.addBatch(sql);
                         }
+                        statement.executeBatch();
+                    } finally {
+                        safeClose(statement);
```
Comment (chengfang) [inline ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java:408]: The `statement` is closed twice: above at line 408, and below line 419. I think we can move the declaration of `statement` into line 400, and remove the close in line 419.

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java [hunk 2]
```diff
@@ -398,13 +398,14 @@ private void checkDatabase() {
                 try {
```
Comment (chengfang) [inline ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java:398]: the nested try block (line 398 and line 401) seems unnecessary. We can combine them into one.

**Comment by petrberan** [comment]
Thank you @chengfang 

Should be resolved now

**Review by chengfang — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @petrberan and @chengfang

**Pull request merged**